src/provider/provider_os_memory_windows.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <Windows.h>
+
+#include <assert.h>
+#include <stdio.h>
+
+#include <umf/providers/provider_os_memory.h>
+
+int os_translate_mem_protection_flags(unsigned protection) {
+    switch (protection) {
+    case UMF_PROTECTION_NONE:
+        return PAGE_NOACCESS;
+    case UMF_PROTECTION_EXEC:
+        return PAGE_EXECUTE;
+    case (UMF_PROTECTION_EXEC | UMF_PROTECTION_READ):
+        return PAGE_EXECUTE_READ;
+    case (UMF_PROTECTION_EXEC | UMF_PROTECTION_READ | UMF_PROTECTION_WRITE):
+        return PAGE_EXECUTE_READWRITE;
+    case (UMF_PROTECTION_EXEC | UMF_PROTECTION_WRITE):
+        return PAGE_EXECUTE_WRITECOPY;
+    case UMF_PROTECTION_READ:
+        return PAGE_READONLY;
+    case (UMF_PROTECTION_READ | UMF_PROTECTION_WRITE):
+        return PAGE_READWRITE;
+    case UMF_PROTECTION_WRITE:
+        return PAGE_WRITECOPY;
+    }
+    fprintf(stderr,
+            "os_translate_mem_protection_flags(): unsupported protection flag: "
+            "%u\n",
+            protection);
+    assert(0);
+    return -1;
+}
+
+void *os_mmap(void *hint_addr, size_t length, int prot) {
```

Comment:
I think we'll need to modify the abstractions a bit (instead of os_mmap, have os_allocate that would do allocation and binding). As far as I understand there is no 'mbind' equivalent on windows. I only know of VirtualAllocExNuma which combines allocation and binding to a specific node (and even then, it's only a hint).

Reply:
Why? We use `hwloc_set_area_membind()`.

Reply:
@igor please add more details about that - we could even sync. However, whatever potential issue we have here should be detected by our tests. Do you have any idea how we can improve them?

Reply:
I'm not 100% sure (I haven't tested it) but from reading the code it seems that hwloc_set_area_membind is not supported on windows. Only hwloc_alloc_membind seems to be supported: https://github.com/open-mpi/hwloc/blob/39fae7e3151fbd677f953f6990acd9eb2a0b9bfb/hwloc/topology-windows.c#L1354

Reply:
As for the tests, I think the existing OS provider tests should cover it - we just need to enable them.

Reply:
The existing OS provider tests require libnuma, but there is no libnuma on Windows ... :-(

Reply:
right.. and even if we write a test for windows, we currently do not have a machine / VM with multiple NUMA nodes. Still, I think it is worth merging this PR to enable building more UMF components on Windows

Reply:
Ok, in that case, can we add an issue to rewrite those tests (perhaps using hwloc)?

Also I think on Windows we will just end up here: https://github.com/oneapi-src/unified-memory-framework/blob/main/src/provider/provider_os_memory.c#L467 Can we add a issue to fix this (remove this check altogether or make it configurable) as well?

Reply:
I ignore this check and added TODO here.